src/test/java/io/stargate/sgv2/jsonapi/api/v1/RangeReadIntegrationTest.java

```diff
@@ -0,0 +1,419 @@
+package io.stargate.sgv2.jsonapi.api.v1;
+
+import static io.restassured.RestAssured.given;
+import static io.stargate.sgv2.common.IntegrationTestUtils.getAuthToken;
+import static net.javacrumbs.jsonunit.JsonMatchers.jsonEquals;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.restassured.http.ContentType;
+import io.stargate.sgv2.jsonapi.config.constants.HttpConstants;
+import io.stargate.sgv2.jsonapi.testresource.DseTestResource;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.ClassOrderer;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestClassOrder;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@QuarkusIntegrationTest
+@QuarkusTestResource(DseTestResource.class)
+@TestClassOrder(ClassOrderer.OrderAnnotation.class)
+public class RangeReadIntegrationTest extends AbstractCollectionIntegrationTestBase {
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Nested
+  @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+  @Order(1)
+  class Find {
+    private final List<Object> testDatas = getDocuments(25);
+
+    @Test
+    @Order(1)
+    public void setUp() {
+      Collections.shuffle(testDatas);
+      insert(testDatas);
+    }
+
+    @Test
+    @Order(2)
+    public void gt() throws Exception {
+      int[] ids = {24, 25};
+      List<Object> testDatas = getDocuments(ids);
+      String json =
+          """
+        {
+          "find": {
+            "filter" : {"userId" : {"$gt" : 23}},
+            "sort" : {"userId" : 1}
+          }
+        }
+        """;
+      JsonNodeFactory nodefactory = objectMapper.getNodeFactory();
+      final ArrayNode arrayNode = nodefactory.arrayNode(testDatas.size());
+      for (int i = 0; i < testDatas.size(); i++)
+        arrayNode.add(
+            objectMapper.readTree(
+                objectMapper
+                    .writerWithDefaultPrettyPrinter()
+                    .writeValueAsString(testDatas.get(i))));
```

Comment:
I _think_ this can be simplified into:

```
arrayNode.add(objectMapper.valueToTree(testDatas.get(i)));
```

where `valueToTree()` is specialized version of `ObjectMapper.convertValue()` which conceptually does "write-then-read" to convert from one type to another.